### PR TITLE
fix-package-selection-broken

### PR DESCRIPTION
**Pull request by solody**
Issues#574 [https://github.com/composer/satis/issues/574](https://github.com/composer/satis/issues/574)

Fix package selection broken when making mirror for https://packages.drupal.org/8

### CODE REVIEW

File: bin/satis [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env php
 <?php
-
+ini_set ("memory_limit","-1");
+error_reporting(E_ERROR | E_WARNING | E_PARSE);
+//报告所有错误
+error_reporting(E_ALL);
+ini_set("display_errors","On");
```
Comment (SvenRtbg) [inline bin/satis:7]: Does this belong into your pull request? Non-englisch comment, increasing memory limit, double setting the error reporting, switching on error display all looks pretty much like leftovers from debugging.

File: src/PackageSelection/PackageSelection.php [hunk 2]
```diff
@@ -30,6 +30,7 @@
 use Composer\Semver\Constraint\EmptyConstraint;
 use Composer\Semver\VersionParser;
 use Composer\Util\Filesystem;
+use PHPUnit\Exception;
```
Comment (SvenRtbg) [inline src/PackageSelection/PackageSelection.php:33]: I think you imported the wrong exception here.

**Comment by SvenRtbg** [comment]
Additionally it would be nice if you'd be able to create any kind of test case for the fix.

**Comment by alcohol** [comment]
Not sure what you are trying to achieve here, but this PR in its current state makes no sense to me. Going to close it.

**Pull request closed without merging**